### PR TITLE
ForeFlight format refactoring

### PR DESCRIPTION
**Pull request by zyv**
I have come to strongly believe that these refactorings could _theoretically_ facilitate the implementation of _similar_ formats by allowing code reuse between format classes.

### CODE REVIEW

File: src/model/formats/foreflight-utils.ts [hunk 1]
```diff
@@ -56,6 +56,17 @@ export class ForeFlightUtils {
       : { type: ChecklistItem_Type.ITEM_PLAINTEXT, prompt: prompt };
   }
 
+  public static possiblyMultilineNoteToChecklistItems(text: string, standalone: boolean): ChecklistItem[] {
```
Comment (rdamazio) [inline src/model/formats/foreflight-utils.ts]: Nit: By reading this function signature, I have no idea what "standalone" means - pick a more descriptive name or add a comment? (alternatively, if you don't think there'll be other uses of the standalone logic, just pass in the in the indent?)

File: src/model/formats/foreflight-writer.ts [hunk 2]
```diff
@@ -100,8 +100,9 @@ export class ForeFlightWriter {
           case ChecklistItem_Type.ITEM_CAUTION:
           case ChecklistItem_Type.ITEM_WARNING: {
             const text = ForeFlightUtils.getChecklistItemPrefix(itemEFIS.type) + itemEFIS.prompt;
-            const [lastItemFF, lastItemEFIS] = accumulator[accumulator.length - 1] || [];
-            if (accumulator.length && lastItemEFIS.indent < itemEFIS.indent) {
+
+            const [lastItemFF, lastItemEFIS] = accumulator.at(-2) ?? [];
```
Comment (rdamazio) [inline src/model/formats/foreflight-writer.ts:106]: Do you really mean to get the second-to-last item here? That's different from the previous logic (and doesn't match the variable names). (or maybe it's just that this change is really part of b9c95ae614703e5a279c525176ffe5f62eef622b ?)

**Review by rdamazio — APPROVED** [review]

File: src/model/formats/foreflight-writer.ts (hunk 2 above)
Comment (zyv) [reply src/model/formats/foreflight-writer.ts:106]: Oh, sorry, this is what you get for faking commit history from a very dirty working copy... I updated the commits to make sense now.

File: src/model/formats/foreflight-utils.ts (hunk 1 above)
Comment (zyv) [reply src/model/formats/foreflight-utils.ts]: I have added a docstring that explains the new tweaked algorithm in more detail. Hopefully this clarifies what "standalone" means in this context. I also updated the test to fix the new behavior (the old one didn't include this edge case, so the tests would pass both before and after my changes).

Previously, the ForeFlight reader read multiline standalone notes as single lines. Of course it will eat the line breaks when exporting. Now it will parse it as a standalone multiline note and the round-trip will work.

|FF|EFIS before|EFIS after|
|--|-----------|----------|
| IMG_1839 | Screenshot 2025-04-09 at 07 21 29 | Screenshot 2025-04-09 at 07 22 52 |

There are still ways to trick the parser (see comment), but I think this was the last practical case that remained unhandled, and the rest cannot be handled without some kind of delimiter, which would make everything even less obvious. Also, I would argue that it just doesn't make sense to format things in a way that would break the new parser.

**Review by rdamazio — APPROVED** [review]

**Comment by rdamazio** [comment]
(ready to merge, just needs a rebase)
